packages/protocol/contracts/stability/Exchange.sol

```diff
@@ -368,9 +406,33 @@ contract Exchange is
     uint256 updatedStableBucket = exchangeRateNumerator.mul(updatedGoldBucket).div(
       exchangeRateDenominator
     );
+
+    uint256 maxStableBucketSize = getStableBucketCap();
+
+    // check if the bucket is bigger
+    uint256 cappedUpdatedStableBucket = Math.min(maxStableBucketSize, updatedStableBucket);
+
+    if (cappedUpdatedStableBucket < updatedStableBucket) {
+      // resize Celo bucket
+      uint256 cappedUpdatedGoldBucket = exchangeRateDenominator.mul((maxStableBucketSize)).div(
+        exchangeRateNumerator
+      );
+      return (cappedUpdatedGoldBucket, cappedUpdatedStableBucket);
+
+    }
+
     return (updatedGoldBucket, updatedStableBucket);
```

Comment:
```suggestion
    // check if the bucket is bigger
    if (updatedStableBucket > maxStableBucketSize) {
      // resize Celo bucket
      uint256 cappedUpdatedGoldBucket = exchangeRateDenominator.mul((maxStableBucketSize)).div(
        exchangeRateNumerator
      );
      return (cappedUpdatedGoldBucket, maxStableBucketSize);
    }

    return (updatedGoldBucket, updatedStableBucket);
```

mixed references to CELO and gold are also slightly confusing

Reply:
yeah, I wanted to have no new references of Gold in the codebase.